crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -602,19 +602,19 @@ impl Context {
         let result_array_id = result_id.to_usize() as u32;
         let result_block_id = BlockId(result_array_id);
 
-        // Initialize the new array with zero values
-        self.initialize_array(result_block_id, len, None)?;
-
-        // Copy the values from the old array into the newly created zeroed array
+        // Initialize the new array with the values from the old array
+        let mut init_values = Vec::new();
```

Comment:
We can use `try_vecmap` here

Reply:
Not really, the error handling will get too cumbersome

Reply:
No, the error handling is the same, you can still use `?`

Reply:
I've switched this over to use `try_vecmap`.